File: aula03/Poligonos/python/python/exemplo.py
from poligonos2 import *

def main():
    triângulo = Triângulo(6,5,5)
    triângulo_retângulo = Triângulo(3,4,5)
    retângulo = Retângulo(3,4)
    quadrado = Quadrado(4)
    pentágono = Polígono([3,3,3,3,3], "Pentágono")

    polígonos = [triângulo, triângulo_retângulo, retângulo, quadrado, pentágono]

    for polígono in polígonos:
        print(polígono)
        print("Perímetro:", polígono.calcula_perímetro())

    print()

    triângulos_e_retângulos = [triângulo, triângulo_retângulo, retângulo, quadrado]

    for objeto in triângulos_e_retângulos:
        print("Área:", objeto.calcula_área())

    nave = ObjetoGeométrico([pentágono, triângulo, triângulo_retângulo, retângulo])
    print("Perímetro da nave:", nave.calcula_perímetro())
    #print("Área da nave:", nave.calcula_área())

if __name__ == "__main__":
    main()

File: aula03/Poligonos/java/java/Exemplo.java
class Exemplo {
    public static void main(String[] args) {
        Polígono triângulo = new Triângulo(6,5,5);
        Polígono triângulo_retângulo = new Triângulo(3,4,5);
        Retângulo retângulo = new Retângulo(3,4);
        Quadrado quadrado = new Quadrado(4);
        int[] lados_pent = {3,3,3,3,3};
        Polígono pentágono = new Polígono(lados_pent, "Pentágono");
        
        Polígono[] polígonos = {triângulo, triângulo_retângulo, retângulo, quadrado, pentágono};

        for (Polígono polígono: polígonos) {
            System.out.println(polígono);
            System.out.println("Perímetro: " + polígono.calcula_perímetro());
        }

        System.out.println("Área do retângulo: " + retângulo.calcula_área());
        System.out.println("Área do quadrado: " + quadrado.calcula_área());
    }    
}
